tests/ipv6.rs

```diff
@@ -0,0 +1,81 @@
+use std::{net::IpAddr, str::FromStr};
```

Comment:
Could you move the address specific test to the file where you define the ip version and iterator? Then the udp and tcp specifics can go in their respective test files.